File: leetcode/0020-valid-parentheses/solution.py
class Solution(object):
    def isValid(self, s):
        if len(s) ==0:
            return False
        c= {
            ")":"(",
            "}":"{",
            "]":"["
        }
        a = []
        for i in s:
            if i in c:
                if len(a) == 0:
                    return False
                else:
                   
                    if c[i] != a[-1]:
                        return False
                    else:
                        a.pop()
            else:
                a.append(i)

        if a:
            return False
        else:
            return True





        """
        :type s: str
        :rtype: bool
        """
        

File: leetcode/0020-valid-parentheses/solution.ts
function isValid(s: string): boolean {
    let stk:string [] = []
    let m = {
        '}':'{'
        ,']':'[',
        ')':'('
    }
    for(let i=0;i<s.length;i++){
        let char = s[i]
        if(char in m){
            if(stk.length ===0){
                return false
            }
            let top = stk[stk.length-1]
            if(top!=m[char]){
                return false

            }
            else{
                stk.pop()
            }
        }
        else{
            stk.push(char)
        }
    }
    return stk.length ===0
};
